### DISCUSSION TITLE
提取单元格四个顶点的坐标信息

### DISCUSSION THREAD

**Discussion by jweden33** [RapidAI/TableStructureRec#100] [Q&A] [answered]
请问大佬 我想提取每个单元格的四个顶点的坐标信息，应该在变量polygons还是ocr_res中提取呢，我没看懂ocr_res的结果，一般ocr_res的格式不应该是[x1, y1, x2, y2, text]吗，为什么我打印了一下输出是这种：[[561.0, 35.0, 760.0, 61.0], [123.0, 42.0, 218.0, 70.0]..],没有包含test的文本信息啊 我想要把提取的四个顶点坐标放在一个长度为8的数组中

**Comment by Joker1212** [comment]
polygons是表格的物理坐标，ocr_res是RapidOcr的结果

**Reply by Joker1212** [reply]
表格还原原理就是ocr的box和表格的box计算包含关系和重合度，所以文本和文本的box都在ocr_res,表格的物理坐标都在polygons

**Reply by jweden33** [reply]
感谢回复！ 明白了，但是还有两个问题需要请教，我把polygons和ocr_res的结果打印出来了，图中的第一行是polygons，第二行是ocr_res
![711](https://github.com/user-attachments/assets/479cef63-a4ec-481f-aea8-5177e6d74b3e)
我没看出来这两个有明显的包含关系，所以想问问这两个都是每个数组中包含四个数值，这四个数值是左上角和右下角的坐标吗，以及每个数值的坐标含义就是[左上横坐标，左上纵坐标，，，]   还有一个问题 您说“文本和文本的box都在ocr_res”可是ocr_res中没有文本，是我理解有误吗

**Reply by Joker1212** [reply, marked as answer]
坐标是xmin,ymin,xmax,ymax，从左到右，从上到下代表min-max的方向，你可以参考两个可视化方法的代码来理解返回值的结构
可视化表格识别框 + 逻辑行列信息
plot_rec_box_with_logic_info(img_path, f"{output_dir}/table_rec_box.jpg", logic_points, polygons)

可视化ocr结果
plot_rec_box(img_path, f"{output_dir}/ocr_box.jpg", ocr_res)

**Reply by jweden33** [reply]
明白了，非常感谢您的耐心解答！
![抱拳了老铁](https://github.com/user-attachments/assets/f3dcde09-1fee-44f9-a134-f286f4f603be)
